flexdll.c

```diff
@@ -357,9 +451,13 @@ static void *find_symbol_global(void *data, const char *name) {
 }
 
 int flexdll_relocate(void *tbl) {
+  err_t * err;
+  err = get_tls_error(TLS_ERROR_RESET_LAST);
+  if(err == NULL) return 0;
```

Comment:
Thinking more about this, shouldn't we reset `err->code = 0` here? Otherwise, the check below in line 460 will fail if this function is called after another function that has set `err->code`.

Going one step further, perhaps when in `TLS_ERROR_RESET_LAST` mode, we should always set `err->code = 0`. Or is there a case where we want to reset one of the error codes, but not the other?

Reply:
Very good point, thank you very much!

I think I ended up with that code because it was not explicitly reset in the original code. That was arguably correct because `flexdll_relocate` is called from two places (if I didn't miss any other):
- from `flexdll_dlopen` where `code` has already been reset,
- from `flexdll_init` where `code` has been set to its initial value 0.
This made me realize that I had forgotten to initialize the values when they are malloc-ed!

So I’ve rewritten the code so that:
- on `TLS_ERROR_RESET`, both `code` and `last_error` are reset (so no `_LAST`),
- the explicit reset of `code` near the call to `get_tls_error(TLS_ERROR_RESET)` are removed,
- the structure is initialized right after `malloc`, just to make sure; the structure _should_ be malloc-ed on a call to one of the initialisation entrypoints, in which case it will be initialised again just a few lines later, but that will ensure that a buggy program calling `dlerror` without a previous call to `dlopen` will get a reliable reasonable behaviour.